src/controllers/credentials/SchemaController.ts

```diff
@@ -1,143 +1,127 @@
-          })
-        }
-        if ((errorMessage.cause.cause, 'CommonInvalidStructure')) {
-          return badRequestError(400, {
-            reason: `schemaId "${schemaId}" has invalid structure.`,
-          })
-        }
+        throw new NotFoundError(getSchemBySchemaId?.resolutionMetadata?.message)
       }
 
-      return internalServerError(500, { message: `something went wrong: ${errorMessage}` })
+      return getSchemBySchemaId
+    } catch (error) {
+      throw ErrorHandlingService.handle(error)
     }
   }
 
   /**
-   * Creates a new schema and registers schema on ledger
-   *
+   * Create schema
    * @param schema
-   * @returns schema
+   * @param notFoundError
+   * @param forbiddenError
+   * @param badRequestError
+   * @param internalServerError
+   * @returns get schema
    */
-  @Example(SchemaExample)
   @Post('/')
-  public async createSchema(
-    @Body()
-    schema: {
-      issuerId: string
-      name: string
-      version: Version
-      attributes: string[]
-      endorse?: boolean
-      endorserDid?: string
-    },
-    @Res() forbiddenError: TsoaResponse<400, { reason: string }>,
-    @Res() internalServerError: TsoaResponse<500, { message: string }>
-  ) {
+  @Example(CreateSchemaSuccessful)
+  public async createSchema(@Body() schema: CreateSchemaInput) {
     try {
       const { issuerId, name, version, attributes } = schema
 
       const schemaPayload = {
-        issuerId: issuerId,
-        name: name,
-        version: version,
+        issuerId,
+        name,
+        version,
         attrNames: attributes,
       }
+      const createSchemaPayload = {
+        schema: schemaPayload,
+        options: {
+          endorserMode: '',
+          endorserDid: '',
+        },
+      }
 
       if (!schema.endorse) {
-        const { schemaState } = await this.agent.modules.anoncreds.registerSchema({
-          schema: schemaPayload,
-          options: {
-            endorserMode: 'internal',
-            endorserDid: issuerId,
-          },
-        })
-
-        const indySchemaId = parseIndySchemaId(schemaState.schemaId)
-        const getSchemaUnqualifiedId = await getUnqualifiedSchemaId(
-          indySchemaId.namespaceIdentifier,
-          indySchemaId.schemaName,
-          indySchemaId.schemaVersion
-        )
-        if (schemaState.state === CredentialEnum.Finished) {
-          schemaState.schemaId = getSchemaUnqualifiedId
-        }
-        return schemaState
+        createSchemaPayload.options.endorserMode = EndorserMode.Internal
+        createSchemaPayload.options.endorserDid = issuerId
       } else {
         if (!schema.endorserDid) {
-          throw new Error('Please provide the endorser DID')
+          throw new BadRequestError(ENDORSER_DID_NOT_PRESENT)
         }
+        createSchemaPayload.options.endorserMode = EndorserMode.External
+        createSchemaPayload.options.endorserDid = schema.endorserDid ? schema.endorserDid : ''
```

Comment:
- We have already added a check above if `schema.endorserDid` is not present
```suggestion
        createSchemaPayload.options.endorserDid = schema.endorserDid
```

Reply:
Yes you are right